src/js/icon_control.js

```diff
@@ -0,0 +1,43 @@
+export class HillshadeControl {
+  constructor({ layerId }) {
+    this._layerId = layerId;
+  }
+
+  _onClick = () => {
+    if (this._map.getLayoutProperty(this._layerId, "visibility") == "none") {
```

Comment:
Astonishingly, my iPhone 8 is able to keep up with the map with the hillshading enabled, but it turns the battery meter into a slow trombone. Should we hide the layer by default on mobile devices, or somehow detect less powerful devices?

Reply:
I upgraded maplibre-gl, the changelog mentions some performance improvements, want to try again?

Reply:
The first render is reasonably quick, but there’s a lot more stuttering while panning than on main. The battery goes down by a few percent after about 20&nbsp;seconds of panning around the Appalachians at zoom level 8. That’s still much better than the majority of ad-laden news websites these days, but the stuttering is a bummer.